node/src/main.rs

```diff
@@ -841,141 +813,21 @@ fn near_networks_as_chains(
         .map(|(chain_id, chain_store, endpoints)| {
             (
                 chain_id.clone(),
-                FirehoseChain {
-                    chain: Arc::new(near::Chain::new(
-                        logger_factory.clone(),
-                        chain_id.clone(),
-                        chain_store,
-                        endpoints.clone(),
-                        metrics_registry.clone(),
-                        Arc::new(NearStreamBuilder {}),
-                    )),
-                    firehose_endpoints: endpoints.clone(),
-                },
+                Arc::new(near::Chain::new(
+                    logger_factory.clone(),
+                    chain_id.clone(),
+                    chain_store,
+                    endpoints.clone(),
+                    metrics_registry.clone(),
+                    Arc::new(NearStreamBuilder {}),
+                )),
             )
         })
         .collect();
 
-    for (chain_id, firehose_chain) in chains.iter() {
-        blockchain_map
-            .insert::<graph_chain_near::Chain>(chain_id.clone(), firehose_chain.chain.clone())
+    for (chain_id, chain) in chains.iter() {
+        blockchain_map.insert::<graph_chain_near::Chain>(chain_id.clone(), chain.clone())
```

Comment:
I hope these changes don't conflict too much with #4408, but we'll see. Let's merge this first as it's bigger.

Reply:
I'm happy to refactor the other PR I think it's not too bad but they touch some of the same things